packages/contracts/contracts/core/permission/PermissionManager.sol

```diff
@@ -161,41 +168,51 @@ contract PermissionManager is Initializable {
         external
         auth(_where, ROOT_PERMISSION_ID)
     {
-        for (uint256 i = 0; i < items.length; i++) {
+        for (uint256 i = 0; i < items.length; ) {
             PermissionLib.ItemSingleTarget memory item = items[i];
 
-            if (item.operation == PermissionLib.Operation.Grant)
+            if (item.operation == PermissionLib.Operation.Grant) {
                 _grant(_where, item.who, item.permissionId);
-            else if (item.operation == PermissionLib.Operation.Revoke)
+            } else if (item.operation == PermissionLib.Operation.Revoke) {
                 _revoke(_where, item.who, item.permissionId);
-            else if (item.operation == PermissionLib.Operation.Freeze)
+            } else if (item.operation == PermissionLib.Operation.Freeze) {
                 _freeze(_where, item.permissionId);
+            }
+
+            unchecked {
+                i++;
+            }
         }
     }
 
     /// @notice Processes bulk items on the permission manager.
     /// @dev Requires that msg.sender has each permissionId on the where.
     /// @param items The array of bulk items to process.
     function bulkOnMultiTarget(PermissionLib.ItemMultiTarget[] calldata items) external {
-        for (uint256 i = 0; i < items.length; i++) {
+        for (uint256 i = 0; i < items.length; ) {
             PermissionLib.ItemMultiTarget memory item = items[i];
 
             // TODO: Optimize
             _auth(item.where, ROOT_PERMISSION_ID);
 
-            if (item.operation == PermissionLib.Operation.Grant)
+            if (item.operation == PermissionLib.Operation.Grant) {
                 _grant(item.where, item.who, item.permissionId);
-            else if (item.operation == PermissionLib.Operation.Revoke)
+            } else if (item.operation == PermissionLib.Operation.Revoke) {
                 _revoke(item.where, item.who, item.permissionId);
-            else if (item.operation == PermissionLib.Operation.Freeze)
+            } else if (item.operation == PermissionLib.Operation.Freeze) {
                 _freeze(item.where, item.permissionId);
-            else if (item.operation == PermissionLib.Operation.GrantWithOracle)
+            } else if (item.operation == PermissionLib.Operation.GrantWithOracle) {
                 _grantWithOracle(
                     item.where,
                     item.who,
                     item.permissionId,
                     IPermissionOracle(item.oracle)
                 );
+            }
+
+            unchecked {
+                i++;
```

Comment:
have u considered using `unchecked{++i;}` as it is cheaper slightly?

Reply:
What do we have `_uncheckedIncrement` for in `UncheckedMath.sol` if we don't use it? 
I would remove it from `UncheckedMath.sol`.